src/lib.rs

```diff
@@ -466,3 +483,443 @@ where
             .merge(transactional_bonsai_storage.tries.db())
     }
 }
+
+/// An alternative to [BonsaiStorage] that does not store data in a trie.
+///
+/// Use this to store data that does not need to be verified but still has to be revertible.
+pub struct RevertibleStorage<ChangeID, DB>
+where
+    DB: BonsaiDatabase,
+    ChangeID: id::Id,
+{
+    db: KeyValueDB<DB, ChangeID>,
+    cache_storage_modified: HashMap<TrieKey, InsertOrRemove<Vec<u8>>>,
+}
+
+impl<ChangeID, DB> RevertibleStorage<ChangeID, DB>
+where
+    DB: BonsaiDatabase,
+    ChangeID: id::Id,
+{
+    /// Create a new revertible storage instance
+    pub fn new(
+        db: DB,
+        config: BonsaiStorageConfig,
+    ) -> Result<Self, BonsaiStorageError<DB::DatabaseError>> {
+        let kvdb = KeyValueDB::new(db, config.into(), None);
+        Ok(Self {
+            db: kvdb,
+            cache_storage_modified: HashMap::new(),
+        })
+    }
+
+    pub fn new_from_transactional_state(
+        db: DB,
+        config: BonsaiStorageConfig,
+        created_at: ChangeID,
+    ) -> Result<Self, BonsaiStorageError<DB::DatabaseError>> {
+        let kvdb = KeyValueDB::new(db, config.into(), Some(created_at));
+        Ok(Self {
+            db: kvdb,
+            cache_storage_modified: HashMap::new(),
+        })
+    }
+
+    /// Insert a new key/value in the storage, overwriting the previous value if it exists.
+    /// If the value already exists it will overwrite it.
+    ///
+    /// > Note: changes will not be applied until the next `commit`
+    pub fn insert(&mut self, identifier: &[u8], key: &BitSlice<u8, Msb0>, value: &[u8]) {
```

Comment:
Yeah as you said in your description identifiers aren't necessary here in this storage. We will let people handle this themself by providing keys that can't collide to the library (maybe add a comment for it)

Reply:
Yep, nuking identifier from `RevertibleStorage` and adding some clarifying comments then :+1: